test/test_hdf5.cpp

```diff
@@ -0,0 +1,156 @@
+//
+// Created by Izzet Yildirim <iyildirim@hawk.iit.edu> on 9/4/24.
+//
+#include <brahma/brahma.h>
+
+#include <iostream>
+#ifdef BRAHMA_ENABLE_HDF5
+#include <hdf5.h>
+
+namespace brahma {
+class HDF5Test : public HDF5 {
+ private:
+  static std::shared_ptr<HDF5Test> instance;
+
+ public:
+  HDF5Test() : api_count(0) {}
+
+  virtual ~HDF5Test() {}
+
+  static std::shared_ptr<HDF5Test> get_instance() {
+    if (instance == nullptr) {
+      instance = std::make_shared<HDF5Test>();
+      HDF5::set_instance(instance);
+    }
+    return instance;
+  }
+
+  size_t api_count;
+
+#if BRAHMA_HDF5_VERSION >= 100800
+  hid_t H5Fcreate(const char* filename, unsigned int flags, hid_t fcpl_id,
+                  hid_t fapl_id) override {
+    api_count++;
+    return HDF5::H5Fcreate(filename, flags, fcpl_id, fapl_id);
+  }
+
+  hid_t H5Dcreate2(hid_t loc_id, const char* name, hid_t type_id,
+                   hid_t space_id, hid_t lcpl_id, hid_t dcpl_id,
+                   hid_t dapl_id) override {
+    api_count++;
+    return HDF5::H5Dcreate2(loc_id, name, type_id, space_id, lcpl_id, dcpl_id,
+                            dapl_id);
+  }
+
+  hid_t H5Screate_simple(int rank, const hsize_t* current_dims,
+                         const hsize_t* maximum_dims) override {
+    api_count++;
+    return HDF5::H5Screate_simple(rank, current_dims, maximum_dims);
```

Comment:
Do not call the actual API. This is just to test interception. No need to do I/O.
```suggestion
    return 0;
```